### PR TITLE
Error message state for destinations.

### PR DESCRIPTION
**Pull request by Nathan-Notifi**
With tooltip content and error message.

in the future, can update this more as the information from the backend becomes more solidified
image

on hover:
image

without tooltip
image

for testing:
- manually update the subscription context original state

### CODE REVIEW

File: packages/notifi-react-card/lib/context/constants.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export type DestinationInputs = {
+  email: string;
+  phoneNumber: string;
+  telegram: string;
+};
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/context/constants.tsx:5]: Record<FormField, string>

File: packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts [hunk 2]
```diff
@@ -138,6 +140,10 @@ export const useNotifiSubscribe: ({
 
       setIsSmsConfirmed(isPhoneNumberConfirmed);
 
+      if (!isPhoneNumberConfirmed) {
+        setPhoneNumberErrorMessage('Invalid number');
+      }
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts:145]: seeing this makes me feel like there's a lot of undefined state that's giong to be tough to maintain. How hard would it be to re-design the `confirmed` state to use the new "destination error" state?

spitballing off top of my head, but maybe something like
```
type RecoverableError = Readonly<{
  type: 'recoverableError';
  label: string;
  onClick: () => void;
}>;

type UnrecoverableError = Readonly<{
  type: 'unrecoverableError';
  label: string;
  tooltip?: string;
}>;

type DestinationError = UnrecoverableError | RecoverableError;
type DestinationErrors = Record<FormField, DestinationError>;
```

**Review by juni-notifi — COMMENTED** [review]
Would you mind seeing if we can consolidate the "unverified" state into the same system?

File: packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts (hunk 2 above)
Comment (Nathan-Notifi) [reply packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts:145]: will take a look!

File: packages/notifi-react-card/lib/context/NotifiFormContext.tsx [hunk 3]
```diff
@@ -24,18 +25,13 @@ export type NotifiFormData = Readonly<{
   setTelegramErrorMessage: (value: string) => void;
```
Comment (marukohao) [inline packages/notifi-react-card/lib/context/NotifiFormContext.tsx:25]: Should the type of the value be DestinationError here?

Comment (Nathan-Notifi) [reply packages/notifi-react-card/lib/context/NotifiFormContext.tsx:25]: for now, no. this is in the formContext so it's handling it a little more simply compared to the subscription context

**Pull request merged**